rpc/rpc-types/src/main/java/com/strategyobject/substrateclient/rpc/types/NumberDecoder.java

```diff
@@ -0,0 +1,22 @@
+package com.strategyobject.substrateclient.rpc.types;
+
+import com.strategyobject.substrateclient.common.utils.HexConverter;
+import com.strategyobject.substrateclient.rpc.core.DecoderPair;
+import com.strategyobject.substrateclient.rpc.core.annotations.AutoRegister;
+import com.strategyobject.substrateclient.rpc.core.decoders.AbstractDecoder;
+import lombok.val;
+
+import java.math.BigInteger;
+
+@AutoRegister(types = Number.class)
+public class NumberDecoder extends AbstractDecoder<Number> {
+    @Override
+    protected Number decodeNonNull(Object value, DecoderPair<?>[] decoders) {
+        val stringValue = (String) value;
+        val hex = stringValue.length() % 2 != 0 ? "0x0" + stringValue.substring(2)
+                : stringValue;
+
+        val number = new BigInteger(HexConverter.toBytes(hex));
```

Comment:
BigInteger can parse hex string. Or is the logic different here?
`val number = new BigInteger(stringValue.substring(2), 16);`

Reply:
Yep, I didn't think about that :)